src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1405,33 +1406,34 @@ uct_dc_mlx5_iface_init_port_affinity(uct_dc_mlx5_iface_t *iface,
         iface->tx.port_affinity = 1;
     }
 
-    dct_port_affinity_idx = UCS_CONFIG_UINT_ENUM_INDEX(config->dct_port_affinity);
-    if (dct_port_affinity_idx == UCT_DC_MLX5_LAG_RESP_PORT_DEFAULT) {
-        iface->rx.port_affinity = 0;
+    dct_affinity = UCS_CONFIG_UINT_ENUM_INDEX(config->dct_affinity);
+    if (dct_affinity == UCT_DC_MLX5_DCT_AFFINITY_DEFAULT) {
+        iface->rx.port_affinity = ib_iface->config.port_num;
         return UCS_OK;
     }
 
-    if (dct_port_affinity_idx == UCT_DC_MLX5_LAG_RESP_PORT_RANDOM) {
+    if (dct_affinity == UCT_DC_MLX5_DCT_AFFINITY_RANDOM) {
         if (md->super.dev.lag_level <= 1) {
-            iface->rx.port_affinity = 0;
+            iface->rx.port_affinity = ib_iface->config.port_num;
         } else {
-            iface->rx.port_affinity = -1;
+            ucs_rand_range(1, md->super.dev.lag_level, &port_affinity);
+            iface->rx.port_affinity = port_affinity;
         }
         return UCS_OK;
     }
 
-    if (config->dct_port_affinity == iface->super.super.super.config.port_num) {
-        iface->rx.port_affinity = 0;
+    if (config->dct_affinity == ib_iface->config.port_num) {
```

Comment:
why need specific check for config->dct_affinity == ib_iface->config.port_num?
lines 1430-1433 should cover this case

Reply:
@yosefe For the case  [dev->lag_level = 0](https://github.com/openucx/ucx/blob/v1.14.0/src/uct/ib/mlx5/dv/ib_mlx5dv_md.c#L1051) & ```config->dct_affinity = ib_iface->config.port_num```,  1430-1433 can't cover it.

Reply:
need to set lag_level=1 if failed to query - not 0

Reply:
@yosefe Done